subprojects/docs/src/samples/templates/project-info-plugin/project-info-plugin/src/main/java/org/gradle/samples/ProjectInfoTask.java

```diff
@@ -0,0 +1,44 @@
+package org.gradle.samples;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.options.Option;
+import org.gradle.api.tasks.options.OptionValues;
+import java.util.EnumSet;
+import java.util.Collection;
+
+public class ProjectInfoTask extends DefaultTask {
+
+    enum Format {
+        PLAIN, JSON
+    }
+
+    private Format format = Format.PLAIN;
+
+    @Option(option = "format", description = "Output format of the project information.")
+    public void setFormat(Format format) {
+        this.format = format;
+    }
+
+    @OptionValues("format")
+    public Collection<Format> getSupportedFormats() {
+        return EnumSet.allOf(Format.class);
+    }
+
+    @TaskAction
+    public void projectInfo() {
+        switch (format) {
+            case PLAIN:
+                getLogger().lifecycle(getProject().getName() + ":" + getProject().getVersion());
+                break;
+            case JSON:
+                getLogger().lifecycle("{\n" +
```

Comment:
❌ I'm actually very sorry to ask this, but it makes more sense to use a `System.out.println` here simply because the whole job of the task is to print to the console. If we would use `-q`, then the task wouldn't print anything with `getLogger().lifecycle`. Sorry for not looking more deeply into what the actual task did.